### PR TITLE
Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by VKTB**
Migrates the CI from Travis to GitHub Actions and updates the README status badge to use that of GitHub Actions. The status badge will start working once this PR gets merged in master, but this is the status badge for this branch to prove that it works:

[Build Status](https://github.com/icatproject/authn.ldap/actions?query=workflow%3A%22CI+Build%22) 

The CI on Travis was not running the integration tests so I configured the GitHub Actions CI to run these from now.
- [ ] Java 8 CI should pass
- [ ] Java 11 CI should pass
- [ ] Should run maven build
- [ ] Should run unit tests
- [ ] Should run integration tests

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        run: |
+          grep payara ~/.bash_profile > payara_path_command
+          source payara_path_command
+          mvn install -DskipTests
+
+      - name: Run Unit Tests
+        run: mvn test -B
+
+      - name: Run Integration Tests
+        run: mvn failsafe:integration-test -B
```
Comment (patrick-austin) [inline .github/workflows/ci-build.yml]:
```suggestion
        run: |
          mvn failsafe:integration-test -B
          mvn failsafe:verify
```

**Review by patrick-austin — CHANGES_REQUESTED** [review]
The integration tests (which weren't running on Travis) have a failure, although this isn't failing the overall workflow - `mvn failsafe:integration-test -B` marks the build as successful regardless of whether the tests pass or fail:
https://github.com/icatproject/authn.ldap/runs/5452098152?check_suite_focus=true#step:17:43
After calling `mvn failsafe:integration-test -B`, we then need to call `mvn failsafe:verify` which checks the result of the failsafe report, and fails the build if the report contains failures (https://stackoverflow.com/questions/12279160/maven-failsafe-plugin-failures-and-build-success). Note that this change should be made to other repos that run integration tests using this method.

Having looked at the failure itself, as far as I can tell it's due to:
https://github.com/icatproject/authn.ldap/blob/de655c7dc130c1a927a46acd04fae1d425646ff7/src/main/resources/run.properties#L1
being unreachable (the error is `class javax.naming.CommunicationException: logon05.fed.cclrc.ac.uk:389` rather than the desired `The username and password do not match`. Maybe this address is no longer up to date, but I do not know what it would need to be changed to as I'm not familiar with this repo. I don't know if you are @VKTB, or if we need someone else's input here.

**Comment by VKTB** [comment]
@patrick-austin  Oh, yes, of course, you are right. I forgot this from the days when I was doing Java.. Thanks for spotting this, I will change it to `mvn failsafe:integration-test failsafe:verify -B` instead so it is one line.

Strangely, that test is passing on my dev machine which is why I never had a look at the output and noticed that there is a failing test. Is it failing for you? I am not familiar with this repo as well so we will definitely need someone else's input on this.

**Comment by VKTB** [comment]
The failing test only passes locally when I am connected to the VPN. There are other reports of RAL-hosted http/https services being unavailable from outside the firewall so it is probably due to this.

**Comment by ajkyffin** [comment]
The RAL LDAP servers will never be accessible form GitHub, so lets disable the integration test and then get this merged. We can try running an OpenLDAP server in Actions to test against at a later date.

**Pull request merged**